src/imageops/colorops.rs

```diff
@@ -72,16 +69,8 @@ where
 /// Invert each pixel within the supplied image.
 /// This function operates in place.
 pub fn invert<I: GenericImage>(image: &mut I) {
-    // TODO why is this not just `for pixel in image.pixels_mut() { pixel.invert(); }`
-    let (width, height) = image.dimensions();
-
-    for y in 0..height {
-        for x in 0..width {
-            let mut p = image.get_pixel(x, y);
-            p.invert();
-
-            image.put_pixel(x, y, p);
-        }
+    for mut pixel in image.pixels() {
+        pixel.2.invert();
```

Comment:
This comment was, unfortunately, slightly outdated and the interpretation is incorrect. This will not invert the pixel within the image as `pixels_mut()` would have done, it will invert a single copy of the color.

Unfortunately, `pixels_mut()` does not work as a generic iterator so it had to be removed and deprecated some time ago.

Reply:
I'm not sure where is the issue here, `pixel.invert()` change the value of the pixel and `.pixels()` will iterate on all the pixels. 
Why this will invert a single copy of the color?

Reply:
`.pixels()` is a read-only iterator over the pixels of an image. Modifying the returned value of `pixel` will be visible locally, but won't get written back to the image itself.

Reply:
This makes sense, so what is the best way to do it? 
By using `borrow_mut()` the change should persist, or just keep the `put_pixel()`

Reply:
You'll need to use `put_pixel`. The change won't persist otherwise, as the code always operates on a copy of the pixel.

Reply:
Okay, I did it, thanks for your help.